data-prepper-api/src/main/java/com/amazon/dataprepper/model/configuration/PipelineModel.java

```diff
@@ -67,11 +75,18 @@ else if (preppers != null) {
 
     public PipelineModel(
             final PluginModel source,
+            final PluginModel buffer,
             final List<PluginModel> processors,
             final List<PluginModel> sinks,
             final Integer workers,
             final Integer delay) {
+        checkArgument(!Objects.isNull(source), "Source must not be null");
```

Comment:
Minor: You could use `Objects.nonNull()` so that you don't have to use negation here